src/backend/distributed/master/master_stage_protocol.c

```diff
@@ -314,15 +310,15 @@ master_append_table_to_shard(PG_FUNCTION_ARGS)
 						  errdetail("Marking this shard placement as inactive")));
 	}
 
-	RESUME_INTERRUPTS();
-
 	/* update shard statistics and get new shard size */
 	newShardSize = UpdateShardStatistics(shardId);
 
 	/* calculate ratio of current shard size compared to shard max size */
 	shardMaxSizeInBytes = (int64) ShardMaxSize * 1024L;
 	shardFillLevel = ((float4) newShardSize / (float4) shardMaxSizeInBytes);
 
+	RESUME_INTERRUPTS();
```

Comment:
Note: I moved this down to avoid erroring out in MultiClientWait if the user cancels, since we want to avoid cancellation here. I separately opened #671, since I don't think we should throw errors from MultiClientWait.